fl4health/utils/losses.py

```diff
@@ -165,156 +162,74 @@ class LossMeterType(Enum):
     ACCUMULATION = "ACCUMULATION"
 
 
-class LossMeter(ABC):
-    def __init__(self) -> None:
-        """
-        A meter to store a list of losses.
-        """
-        self.losses_list: List[Losses] = []
-
-    def get_type(self) -> LossMeterType:
-        """
-        Returns the type of this loss meter
-
-        Raises:
-            NotImplementedError: To be implemented by child classes.
-        """
-        raise NotImplementedError
-
-    @abstractmethod
-    def update(self, losses: Losses) -> None:
-        """
-        Appends loss to list of losses.
-
-        Args:
-            losses (Losses): A losses object with checkpoint, backward and additional losses.
-
-        Raises:
-            NotImplementedError: To be implemented by child classes.
-        """
-        raise NotImplementedError
-
-    @abstractmethod
-    def clear(self) -> None:
-        """
-        Reset the meter by re-initializing losses_list to be empty
-
-        Raises:
-            NotImplementedError: To be implemented by child class.
-        """
-        raise NotImplementedError
-
-    def compute(self) -> Losses:
-        """
-        Compute average of current list of losses if non-empty.
-
-        Returns:
-            Losses: New Losses object with average of losses in losses_list.
-        """
-        assert len(self.losses_list) > 0
+LossesType = TypeVar("LossesType", bound=Losses)
 
-        # assume the type of the first loss object is the type of all loss objects
-        if isinstance(self.losses_list[0], EvaluationLosses):
-            return EvaluationLosses.aggregate(self)
 
-        return TrainingLosses.aggregate(self)
-
-    @classmethod
-    def get_meter_by_type(cls, meter_enum: LossMeterType) -> LossMeter:
+class LossMeter(Generic[LossesType]):
+    def __init__(self, loss_meter_type: LossMeterType, losses_type: type[Losses]) -> None:
```

Comment:
I might be wrong, but can `losses_type` here actually be annotated losses_type: LossesType? That is, it must be the same time as the generic for LossMeter?

Reply:
good point, thanks